src/com/facebook/buck/plugin/BuckPluginManagerFactory.java

```diff
@@ -34,6 +40,24 @@ public static PluginManager createPluginManager() {
           protected ExtensionFinder createExtensionFinder() {
             return new BuckExtensionFinder();
           }
+
+          @Override
+          protected CompoundPluginDescriptorFinder createPluginDescriptorFinder() {
+            CompoundPluginDescriptorFinder finder = new CompoundPluginDescriptorFinder();
+            finder.add(new ManifestPluginDescriptorFinder());
+            return finder;
+          }
+
+          @Override
+          protected PluginRepository createPluginRepository() {
+            try {
+              return new DefaultPluginRepository(
```

Comment:
Why not `new DefaultPluginRepository(getPluginsRoot(), isDevelopment()`?

Reply:
`getPluginsRoot()` is what returns `plugins` path by default which is exactly what we do not want.

Also its strange to run it in development mode as we do not know what other weirdness that will turn on

Reply:
Could you instead override `createPluginsRoot()` and use `buck-plugins` name instead? Then you can use `new DefaultPluginRepository(getPluginsRoot(), isDevelopment())`

Reply:
Im ok with that approach :)